config/environment.rb

```diff
@@ -3,3 +3,9 @@
 
 # Initialize the Rails application.
 Rails.application.initialize!
+
+queue_host = ENV["QUEUE_HOST"] || "localhost"
+queue_port = ENV["QUEUE_PORT"] || 9092
+
+service_order_listener = ServiceOrderListener.new(:host => queue_host, :port => queue_port)
+service_order_listener.run
```

Comment:
@eclarizio For Sanity testing that consumer groups work, we should test with two or more threads as part of an consumer group and make sure only one of them accesses it. Its also possible that if the messages come in too quickly from the topology service these threads might update messages in the incorrect order.

Reply:
@eclarizio I think I would rather see this in an initializer.  Normally external services get configured there, and rails only configuration lives in `environment.rb`.

Reply:
In regards to [this question](https://github.com/ManageIQ/catalog-api/pull/134#issuecomment-471791640) if we move it into an initializer you could also only allow it to turn on in `production`. I do like the idea of a separate bin file a lot.